.github/workflows/tests.yml

```diff
@@ -311,6 +312,7 @@ jobs:
         run: sudo chmod -R g-w,o-w /home/linuxbrew/.linuxbrew/Homebrew
 
       - name: Run brew tests
+        if: github.event_name == 'pull_request' || matrix.name != 'tests (online)'
```

Comment:
I think it might actually make more sense to flip this condition.

Running online tests only in the merge queue will make it less likely we hit rate limits if we also limit to one concurrent merge group.

Reply:
@reitermarkus I'm less concerned about rate limits here than the job being flaky and that being annoying before merge. If it flakes when in the merge queue, it'll get kicked out and you need to try and add and rerun all CI again. If it flakes before the merge queue, you can manually rerun.

Reply:
But what does flaky mean exactly? I think we already retry online tests multiple times to make them more robust, so the only thing that can still fail them is the rate limit.

Can you point to an online test that is flaky for a different reason?

Reply:
@reitermarkus Flaky meaning will fail intermittently and we've been unable to fix that. That applies with pretty much all our online tests.

It's rare for them to fail consistently outside of the rate limit but e.g. third-party services being down/flaky will make them fail/flake too.

Reply:
My only concern is we don't get the benefit of merge queues when two PRs modify code that needs network access. I guess we'll cross that bridge if we ever get there and add offline tests.

Reply:
@reitermarkus yeh, agreed 👍🏻